core/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2ClientAuthenticationConverterTests.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.web;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.server.authorization.authentication.OAuth2ClientAuthenticationToken;
+import java.util.Base64;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Patryk Kostrzewa
+ */
+public class OAuth2ClientAuthenticationConverterTests {
+
+	private OAuth2ClientAuthenticationConverter converter;
+
+	@Before
+	public void setup() {
+		converter = new OAuth2ClientAuthenticationConverter();
+	}
+
+	@Test
+	public void shouldConvertToAuthenticationToken() {
```

Comment:
Please see existing tests in `spring-security-oauth2-client` module and follow the method naming pattern and general test style. eg. `DefaultOAuth2AuthorizationRequestResolverTests`, `OAuth2AuthorizationCodeAuthenticationProviderTests`, etc.